sdk/src/Services/DynamoDBv2/Custom/DataModel/IDynamoDBContextBuilder.cs

```diff
@@ -0,0 +1,49 @@
+﻿/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ * 
+ *  http://aws.amazon.com/apache2.0
+ * 
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+using System;
+
+namespace Amazon.DynamoDBv2.DataModel
+{
+    /// <summary>
+    /// Interface for a builder that constructs a <see cref="DynamoDBContext"/>
+    /// </summary>
+    public interface IDynamoDBContextBuilder : IDisposable
+    {
+        /// <summary>
+        /// Sets the DynamoDB client to be used by the <see cref="DynamoDBContext"/> that is constructed
+        /// </summary>
+        /// <param name="client"><see cref="IAmazonDynamoDB"/> to use to access DynamoDB.</param>
+        IDynamoDBContextBuilder SetDynamoDBClient(IAmazonDynamoDB client);
```

Comment:
Instead of these 2 set methods for client and region it might be better to have method to add a client factory method. Having to methods makes me think I could set both which would be undefined behavior especially of the region being set is different then the client past in. Also if we have these 2 do we need to add a third for setting the Client config.

Something like:
```
IDynamoDBContextBuilder WithDynamoDBClient(Func<IAmazonDynamoDBClient> factory);
```

If they don't ever call this then you use the internal factory of creating a client like we do today when a client is not provided. Call that out in the docs.